nion/swift/ExportDialog.py

```diff
@@ -239,45 +239,133 @@ def export_clicked(display_items: typing.Sequence[DisplayItem.DisplayItem], view
+                self.width_model.value = new_width
+                self.width_value_line_edit.background_color = "#d3d3d3"
+                self.height_value_line_edit.background_color = "white"
+        except ValueError:
+            pass  # Handle the case where the input is not a valid integer
+    def unit_changed(self, widget: Declarative.UIWidget, current_index: int) -> None:
+        width_px, height_px = self.convert_to_pixels(
+            self.width_model.value, self.height_model.value, self.previous_unit_index)
+        new_width, new_height = self.convert_from_pixels(width_px, height_px, current_index)
+        if current_index == 2:
+            new_width = round(new_width)
+            new_height = round(new_height)
+        self.width_model.value = new_width
+        self.height_model.value = new_height
+        self.previous_unit_index = current_index
+    def convert_to_pixels(self, width: typing.Optional[float], height: typing.Optional[float], unit_index: typing.Optional[int]) -> typing.Tuple[int, int]:
+        dpi = 96
+        if width is None or height is None:
+            return 0, 0
+        else:
+            width = float(width)
+            height = float(height)
+        if unit_index == 0:  # Inches
+            return int(width * dpi), int(height * dpi)
+        elif unit_index == 1:  # Centimeters
+            return int(width * dpi / 2.54), int(height * dpi / 2.54)
+        return int(round(width)),int(round(height)) # Pixels
+    def convert_from_pixels(self, width_px: typing.Optional[float], height_px: typing.Optional[float], unit_index: typing.Optional[int]) -> typing.Tuple[float, float]:
+        dpi = 96
+        if width_px is None or height_px is None:
+            return 0, 0
+        else:
+            width_px = float(width_px)
+            height_px = float(height_px)
+        if unit_index == 0:  # Inches
+            return width_px / dpi, height_px / dpi
+        elif unit_index == 1:  # Centimeters
+            return width_px * 2.54 / dpi, height_px * 2.54 / dpi
+        return width_px, height_px  # Pixels
+    def get_dimensions_in_pixels(self) -> typing.Tuple[int, int]:
+        return self.convert_to_pixels(
+            self.width_model.value, self.height_model.value, self.unit_model.value)
+
+
     def close(self) -> None:
         pass
 
-
 class ExportSVGDialog:
-
     def __init__(self, document_controller: DocumentController.DocumentController, display_item: DisplayItem.DisplayItem) -> None:
         super().__init__()
-
         self.__document_controller = document_controller
-
         u = Declarative.DeclarativeUI()
-
         if display_item.display_data_shape and len(display_item.display_data_shape) == 2:
-            display_size = Geometry.IntSize(height=4, width=4)
+            display_size = Geometry.IntSize(height=display_item.display_data_shape[1],
+                                            width=display_item.display_data_shape[0])
```

Comment:
The dimension of `display_size` are swapped. We use the numpy convention: height is `[0]` and width is `[1]`. Try exporting a 10x100 and 100x10 data item.